src/main/webapp/user.js

```diff
@@ -58,11 +91,30 @@ class User {
     registrationLink.textContent = "Register Organization";
     navBar.appendChild(registrationLink);
 
-    if(this.isMaintainer) {
+    if (this.isMaintainer) {
       const organizationsLink = document.createElement("a");
       organizationsLink.setAttribute("href", "organizations.html");
       organizationsLink.textContent = "Organizations";
       navBar.appendChild(organizationsLink);
+
+      const addMaintainerPopup = this.createAddMaintainerPopup();
+      const addMaintainerLabel = document.createElement("a");
+      addMaintainerLabel.textContent = "Add Maintainer";
+      addMaintainerLabel.setAttribute("id", "add-maintainer-label");
+      addMaintainerLabel.addEventListener("click", () => {
+        const addMaintainerPopupBackground = document.createElement("div");
+        addMaintainerPopupBackground.setAttribute("id", "modal-popup-background");
+        addMaintainerPopupBackground.classList.add("add-maintainer-popup-modal-background");
+        addMaintainerPopupBackground.appendChild(addMaintainerPopup);
+
+        const mainContent = document.getElementById("help-near-me-content") ? document.getElementById("help-near-me-content") : document.getElementById("organizations-content");
```

Comment:
This seems weird to me. Why add it to the mainContent and not document.body ? As is, you are leaking what page is hosting this popup into the logic here. What happens when/if you have a new page?

Reply:
I agree! I fixed it to prepend to the body of the document instead of to specific pages.